src/compiler/checker.ts

```diff
@@ -5765,12 +5765,13 @@ namespace ts {
             return type.modifiersType;
         }
 
+        function isPartialMappedType(type: Type) {
+            return getObjectFlags(type) & ObjectFlags.Mapped && !!(<MappedType>type).declaration.questionToken;
+        }

```

Comment:
As we discussed offline, that example doesn't apply because a constraint being all-optional (e.g. `MyInterface`) doesn't guarantee that `T` itself will be all-optional.